util/topgen/intermodule.py

```diff
@@ -829,3 +855,38 @@ def im_portname(obj: OrderedDict, suffix: str = "") -> str:
         suffix_s = "_o" if obj["act"] == "rsp" else "_i"
 
     return "{signame}{suffix}".format(signame=obj["name"], suffix=suffix_s)
+
+
+def get_dangling_im_def(objs: OrderedDict) -> str:
+    """return partial inter-module definitions
+
+    Dangling intermodule connections happen when a one-to-N assignment
+    is not fully populated.
+
+    This can result in two types of dangling:
+    - outgoing requests not used
+    - incoming responses not driven
+
+    The determination of which category we fall into follows similar rules
+    as those used by im_netname.
+
+    When the direction of the net is the same as the active direction of the
+    the connecting module, it is "unused".
+
+    When the direction of the net is opposite of the active direction of the
+    the connecting module, it is "undriven".
+
+    As an example, edn is defined as "rsp" of a "req_rsp" pair. It is also used
+    as the "active" module in inter-module connection. If there are not enough
+    connecting modules, the 'req' line is undriven, while the 'rsp' line is
+    unused.
```

Comment:
It appears that the big picture here is to stub unused buses, such as in the case of EDN. It seems like the number of ports parameter could be used and avoid all of this stubbing python code. I am sure I am missing something, but it could simple be a case of increasing the parameter as new endpoints become connected.

Reply:
@mwbranstad yes absolutely.  If we were to change `edn` to template module, such as `flash_ctrl` or `otp_ctrl` (these are different styles of templating), we can always match the number of ports on `edn` to exactly the number of connections.

As it is right now though, `edn` is not templated, and the number of ports is a parameter stored in generated package file (I'm guessing it also impacts the register file?) So there's not an easy way to manipulate that.  Being able to parameterize our regfile has long been an issue.

Reply:
o @mwbranstad i just took a look at `edn`, and i think i was mistaken.  Am i correct to say the number of connections edn supports DOES NOT change the regfile? Because if that's the case, pass a parameter through to edn directly via toplevel (this is something we support) might indeed be easier.

Reply:
o wait.. i misspoke yet again... even though it doesn't impact the regfile, it does impact the inter-signal definition inside edn.hjson.. so yeah... the way for us to get around this problem would be templating in this case.  But I'm not sure if that's something you want to support.

Reply:
@tjaychen yeah it supports the number of endpoint ports by a parameter, but if that is an issue with this connection system I guess some stubbing scheme is needed. It seems that if changing a parameter also impacts the edn.hjson file, then it is two coordinated changes versus just the parameter value. Maybe there is still more to the interconnection flow.

Reply:
In the long run it would be nice if important system infrastructure is automatically parameterized.
E.g. for the PLIC we also automatically figure out how many interrupts there are, and pass this parameter into the templated design below.

However, for now I think this works fine as well.
Moreover, the mechanism that @tjaychen has added here is not specific to EDN - it can be used to stub off other 1:N connections automatically as well.

Reply:
@msfschaffner when you say automatically parameterized, is there any option here other than templating? 
The width of the edn interface is declared inside edn.hjson, so there's no way for normal RTL to touch that without us modifying the hjson on the fly. 

Of course maybe we can think of a way to re-do this whole system, but that's probably a different topic.